### PR TITLE
Add doc for NavigationMesh connections and NavigationDebug

### PR DESCRIPTION
**Pull request by smix8**
Adds documentation how to merge and / or connect different navigationmeshes
Adds documentation how for the available navigation debug tools.

### CODE REVIEW

File: tutorials/navigation/navigation_connecting_navmesh.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+.. image:: img/navigation_vertex_merge.png
+
+The same is true for multiple NavigationPolygon resources. As long as their 
+outline points overlap exactly the NavigationServer will merge them.
+NavigationPolygon outlines must be from different NavigationPolygon resources to connect.
+
+Overlapping or intersecting outlines on the same NavigationPolygon 
+will fail the navigation mesh creation. Overlapping or intersecting 
+outlines from different NavigationPolygons will often fail the 
+navigation region connections on the NavigationServer and should be avoided.
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_connecting_navmesh.rst:18]: Feels like a work may be missing hereß "fail _at_ the connections"? I dunno, maybe I'm just missing something :)

File: tutorials/navigation/navigation_connecting_navmesh.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+.. image:: img/navigation_vertex_merge2.png
+
+.. warning::
+
+    Exactly means exactly for the vertex position merge. Small float errors 
+    that happen quite regularly with imported meshes will prevent a successful vertex merge.
+
+Alternatively NavigationMesh are not merged but still considered as ``connected`` by
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_connecting_navmesh.rst:27]:
```suggestion
Alternatively ``NavigationMesh``s are not merged but still considered as ``connected`` by 
```

**Review by mhilbrunner — COMMENTED** [review]
See comments!

File: tutorials/navigation/navigation_connecting_navmesh.rst (hunk 1 above)
Comment (smix8) [reply tutorials/navigation/navigation_connecting_navmesh.rst:18]: ... will often fail to create the navigation region `edge connections` on the NavigationServer ...

The `edge connections` are their own thing build by the server behind the scene. Merges the free edges of a navigationregion navmesh with other free navigationregion navmesh edges by proximity / angle.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks!

**Pull request merged**